### ISSUE TITLE
Questions regarding the access to sensitive attributes and the usage of labels.

### ISSUE THREAD

**Issue by AllenZ01** [ZzoomD/FairINV#2]
Hi,

Thanks for sharing your code. I’ve been going through it and had a few questions that I hope you could help clarify:

1. **Access to Sensitive Attributes**: The paper mentions that FairINV does not require access to sensitive attributes (The 12th line of the third paragraph in the Introduction),
>Correspondingly, our first challenge is to design a fairness framework that achieves fairness w.r.t. various sensitive attributes in a single training session **without accessing the sensitive attribute**, as shown in Figure 1(b). 

but I couldn’t find any step in the code that removes sensitive attributes. It seems that the input features for the pre-trained `ref_model` and `SAP` module, and the overall `FairINV` all include sensitive attributes.

2. **Usage of Labels During Training**: It looks like all node labels are being used in the forward function of the `SAP` module. However, I thought that labels for the validation and test sets should not be accessible during training. Is there any misunderstanding here?

3. **The `IRMLoss` Class**: I wanted to check if the `IRMLoss` class defined in `fairinv.py` is unused, as I didn’t find any references to it in the code.

Thanks again for sharing your code, and I’d appreciate any clarification you can provide to help resolve my misunderstanding.

**Comment by ZzoomD** [comment]
Hi, AllenZ01,

Thank you for your attention.

**For the first question**, our method does not require sensitive attributes because the training of SAP (Algorithm 1, lines 2-13) does not need to know the sensitive attributes. The partitioning result of SAP can serve as the sensitive attribute, which is used for the next step of training (Algorithm 1, lines 16-23). In our experiments, we use sensitive attributes to make a fair comparison with existing methods, which cannot be used without sensitive attributes. You can directly remove the columns related to sensitive attributes when loading the data.

**For the second question**, we only use the training set for model parameter optimization throughout the training process, including SAP (you can check the sens_partition function). We only use the validation set to select the optimal model and do not use it to calculate the loss. Therefore, our training process only uses the training set.

**For the third question**, we did not use the IRMLoss class in the code. We directly wrote the IRM objective mentioned in the paper into the sens_partition function (line 174 of the [fairinv.py](http://fairinv.py/) file). Thank you for pointing this out, and we will remove the IRMLoss class.

**Comment by AllenZ01** [comment]
Hi, ZzoomD,

Thank you for your detailed response. My questions have been resolved. 

Thank you, and best wishes.
